data-service/src/unit-tests/java/uk/gov/gchq/palisade/service/data/service/SimpleDataServiceTest.java

```diff
@@ -18,87 +18,207 @@
-            .user(entity.getUser())
-            .resource(entity.getLeafResource())
-            .context(entity.getContext())
-            .rules(entity.getRules());
+            .user(USER)
+            .resource(LEAF_RESOURCE)
+            .context(CONTEXT)
+            .rules(RULES);
+
+    final AuthorisedData authorisedData = AuthorisedData.Builder.create()
+            .withResource(LEAF_RESOURCE)
+            .withUser(USER)
+            .withContext(CONTEXT)
+            .withRules(RULES);
+
+    final String testResponseMessage = "test response for data request";
+
+    final ResponseWriter responseWriter = new ResponseWriter() {
+        final String testData = testResponseMessage;
+
+        @Override
+        public void close() throws IOException {
+
+        }
 
+        @Override
+        public ResponseWriter write(final OutputStream outputStream) throws IOException {
+
+            InputStream testInputStream = new ByteArrayInputStream(testData.getBytes());
+            testInputStream.transferTo(outputStream);
+            testInputStream.close();
+            return this;
+        }
+    };
+
+    final ResponseWriter errorProneResponseWriter = new ResponseWriter() {
+        final String testData = testResponseMessage;
+
+        @Override
+        public void close() {
+
+        }
+
+        @Override
+        public ResponseWriter write(final OutputStream outputStream) throws IOException {
+            throw new IOException("Something went wrong");
+
+        }
+    };
+
+    final DataReaderResponse dataReaderResponse = new DataReaderResponse()
+            .message("test message")
+            .writer(responseWriter);
+
+    final DataReaderResponse errorDataReaderResponse = new DataReaderResponse()
+            .message("test message")
+            .writer(errorProneResponseWriter);
+
+    /**
+     * Test for {@link SimpleDataService#authoriseRequest(DataRequest)}.  If the request is found to be
+     * authorised, the response will be the relevant information needed to proceed with the request. This will be in the
+     * form of a {@code DataReaderRequest}.
+     */
     @Test
-    void testAuthoriseRequestDelegatesToPersistence() {
-        // Given the persistence layer is mocked to return the test data
-        Mockito.when(persistenceLayer.getAsync(Mockito.anyString(), Mockito.anyString()))
-                .thenReturn(CompletableFuture.completedFuture(Optional.of(entity)));
+    void testAuthoriseRequestWithAValidRequest() {
+        // Given
+        when(persistenceLayer.getAsync(any(), any()))
+                .thenReturn(CompletableFuture.completedFuture(Optional.of(authorisedEntity)));
 
         // When
-        CompletableFuture<Optional<DataReaderRequest>> authorisedRequest = simpleDataService.authoriseRequest(dataRequest);
+        CompletableFuture<AuthorisedData> dataReaderRequestCompletableFuture = simpleDataService.authoriseRequest(dataRequest);
```

Comment:
I prefer the simpler name, this is only really spelling out the type signature, and incorrectly so
```suggestion
        CompletableFuture<AuthorisedData> authorisedRequest = simpleDataService.authoriseRequest(dataRequest);
```

Reply:
Typo in the suggested change, but I got ther idea and made the change in the code.